File: greenpepper-open/extensions-external/php/src/test/php/bank/AccountType.php
<?php

include_once dirname(__FILE__).'/WithdrawType.php';
include_once dirname(__FILE__).'/Money.php';

class AccountType 
{ 
 
  public static $SAVINGS;
  public static $CHECKING;
  
  private $description; 
 
  private $limits = array(); 
  private $fees = array(); 
 
  private function AccountType($description, Money $limitBankMachine = NULL, 
            Money $limitInteracMachine = NULL, Money $limitPersonalCheck = NULL, 
            Money $feesBankMachine = NULL, Money $feesInteracMachine = NULL, Money $feesPersonalCheck = NULL) 
  { 
    $this->description = $description; 
    $this->limits[WithdrawType::$ATM->getId()] = $limitBankMachine; 
    $this->limits[WithdrawType::$INTERAC->getId()] = $limitInteracMachine; 
    $this->limits[WithdrawType::$PERSONAL_CHECK->getId()] = $limitPersonalCheck; 
 
    $this->fees[WithdrawType::$ATM->getId()] = $feesBankMachine; 
	$this->fees[WithdrawType::$INTERAC->getId()] = $feesInteracMachine; 
	$this->fees[WithdrawType::$PERSONAL_CHECK->getId()] = $feesPersonalCheck; 
  } 
  
  public function toString() 
  { 
    return $this->description; 
  } 
 
  public function limitFor(WithdrawType $withdrawType) 
  { 
    return $this->limits[$withdrawType->getId()]; 
  } 
  
  public function feesFor(WithdrawType $withdrawType) 
  { 
    return $this->fees[$withdrawType->getId()]; 
  } 
 
  public static function isNoLimit(Money $limit = NULL) 
  { 
    return $limit == null; 
  } 
  
  public static function init() {
	AccountType::$SAVINGS = new AccountType("savings", Money::parse(200), Money::parse(200), null, Money::parse(1.5), Money::parse(2.5), Money::parse(0.5));
	AccountType::$CHECKING = new AccountType("checking", Money::parse(200), Money::parse(200), null, Money::parse(0), Money::parse(2.50), Money::parse(0)); 
  }
} 

AccountType::init();

File: greenpepper-open/samples/src/main/java/com/greenpepper/samples/application/bank/AccountType.java
package com.greenpepper.samples.application.bank;

import java.util.EnumMap;

/**
 * <p>AccountType class.</p>
 *
 * @author oaouattara
 * @version $Id: $Id
 */
public enum AccountType
{
	SAVINGS ("savings", Money.$(200f), Money.$(200f), null,  Money.$(1.5f), Money.$(2.5f), Money.$(0.5f)),
	CHECKING ("checking", Money.$(200f), Money.$(200f), null, Money.$(0), Money.$(2.50f), Money.$(0));

	private final String description ;

	private EnumMap<WithdrawType, Money> limits;
	private EnumMap<WithdrawType, Money> fees;

	private AccountType (String description, Money limitBankMachine,
            Money limitInteracMachine, Money limitPersonalCheck,
            Money feesBankMachine, Money feesInteracMachine, Money feesPersonalCheck)
	{
		this.description = description;
		limits = new EnumMap<WithdrawType, Money>(WithdrawType.class);

		limits.put(WithdrawType.ATM, limitBankMachine);
		limits.put(WithdrawType.INTERAC,  limitInteracMachine );
		limits.put(WithdrawType.PERSONAL_CHECK, limitPersonalCheck );

		fees = new EnumMap<WithdrawType, Money>(WithdrawType.class);
		fees.put(WithdrawType.ATM,  feesBankMachine );
		fees.put(WithdrawType.INTERAC, feesInteracMachine );
		fees.put(WithdrawType.PERSONAL_CHECK, feesPersonalCheck );

	}

	/** {@inheritDoc} */
	@Override
	public String toString() {
		return description;
	}

	/**
	 * <p>limitFor.</p>
	 *
	 * @param withdrawType a {@link com.greenpepper.samples.application.bank.WithdrawType} object.
	 * @return a {@link com.greenpepper.samples.application.bank.Money} object.
	 */
	public Money limitFor(WithdrawType withdrawType)
	{
		return limits.get(withdrawType);
	}

	/**
	 * <p>feesFor.</p>
	 *
	 * @param withdrawType a {@link com.greenpepper.samples.application.bank.WithdrawType} object.
	 * @return a {@link com.greenpepper.samples.application.bank.Money} object.
	 */
	public Money feesFor(WithdrawType withdrawType)
    {
		return fees.get(withdrawType);
	}

	/**
	 * <p>isNoLimit.</p>
	 *
	 * @param limit a {@link com.greenpepper.samples.application.bank.Money} object.
	 * @return a boolean.
	 */
	public static boolean isNoLimit(Money limit)
	{
		return limit == null;
	}

}
